File: week6/Problem_Set6/sentimental-credit/credit.py
# TODO


def main():
    while True:
        n = input("Number: ")
        count = len(n)
        n = int(n)
        if n > 0:
            break

    if count != 13 and count != 15 and count != 16:
        print("INVALID")
        return 0

    if isValid(n):
        while n >= 100:
            n = int(n / 10)
        if (n == 37 or n == 34) and count == 15:
            print("AMEX")
        elif n >= 51 and n <= 55 and count == 16:
            print("MASTERCARD")
        elif int(n / 10) == 4:
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")
    return 0


def isValid(n):
    sumMul2 = 0
    sum = 0
    while n != 0:
        sum += int(n % 10)
        n = int(n / 10)
        if n % 10 * 2 >= 10:
            sumMul2 += int((n % 10 * 2) % 10 + (n % 10 * 2) / 10)
        else:
            sumMul2 += int(n % 10 * 2)
        n = int(n / 10)

    res = sum + sumMul2
    if res % 10 == 0:
        return True
    else:
        return False


main()

File: week1/Problem_Set1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int Count(long long int n)
{
    int count = 0;
    while (n != 0)
    {
        n /= 10;
        count++;
    }
    return count;
}

bool isValid(long long int n)
{
    int sumMul2 = 0;
    int sum = 0;
    while (n != 0)
    {
        sum += n % 10;
        n /= 10;
        if (n % 10 * 2 >= 10)
        {
            sumMul2 += (n % 10 * 2) % 10 + (n % 10 * 2) / 10;
        }
        else
            sumMul2 += n % 10 * 2;
        n /= 10;
    }
    int res = sum + sumMul2;
    if (res % 10 == 0)
        return true;
    else
        return false;
}

int main(void)
{
    long long int n;
    do
    {
        n = get_long("Number: ");
    }
    while (n <= 0);

    int cnt = Count(n);
    if (cnt != 13 && cnt != 15 && cnt != 16)
    {
        printf("INVALID\n");
    }
    else
    {
        if (isValid(n))
        {
            while (n >= 100)
            {
                n /= 10;
            }
            if ((n == 37 || n == 34) && cnt == 15)
            {
                printf("AMEX\n");
            }
            else if (n >= 51 && n <= 55 && cnt == 16)
            {
                printf("MASTERCARD\n");
            }
            else if (n / 10 == 4)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
}